src/commands/data/import/resume.ts

```diff
@@ -0,0 +1,193 @@
+          get: (data): string | undefined => {
+            if (data?.numberRecordsProcessed) {
+              return data.numberRecordsProcessed.toString();
+            }
+          },
+        },
+        {
+          stage: 'Processing the job',
+          label: 'Successful records',
+          type: 'dynamic-key-value',
+          get: (data): string | undefined => {
+            const numberRecordsFailed = data?.numberRecordsFailed ?? 0;
+
+            if (data?.numberRecordsProcessed) {
+              return (data.numberRecordsProcessed - numberRecordsFailed).toString();
+            }
+          },
+        },
+        {
+          stage: 'Processing the job',
+          label: 'Failed records',
+          type: 'dynamic-key-value',
+          get: (data): string | undefined => {
+            const numberRecordsFailed = data?.numberRecordsFailed ?? 0;
+
+            if (data?.numberRecordsProcessed) {
+              return numberRecordsFailed.toString();
+            }
+          },
+        },
+      ],
+      postStagesBlock: [
+        {
+          label: 'Status',
+          type: 'dynamic-key-value',
+          bold: true,
+          get: (data) => data?.state,
+        },
+        {
+          label: 'Job Id',
+          type: 'dynamic-key-value',
+          bold: true,
+          get: (data) =>
+            data?.id &&
+            terminalLink(
+              data.id,
+              `${
+                conn.getAuthInfoFields().instanceUrl as string
+              }/lightning/setup/AsyncApiJobStatus/page?address=${encodeURIComponent(`/${data.id}`)}`
+            ),
+        },
+      ],
+    });
+
+    ms.skipTo('Processing the job');
+
+    const job = conn.bulk2.job('ingest', {
+      id: resumeOpts.jobInfo.id,
+    });
+
+    job.on('inProgress', (res: JobInfoV2) => {
+      ms.goto('Processing the job', res);
+    });
+
+    try {
+      await job.poll(5000, flags.wait.milliseconds);
+
+      const jobInfo = await job.check();
+
+      // send last data update so job status/num. of records processed/failed represent the last update
+      ms.goto('Processing the job', jobInfo);
+
+      if (jobInfo.numberRecordsFailed) {
+        ms.stop('failed');
+        // TODO: replace this msg to point to `sf data bulk results` when it's added (W-12408034)
+        throw messages.createError('error.failedRecordDetails', [
+          jobInfo.numberRecordsFailed,
+          conn.getUsername(),
+          job.id,
+        ]);
+      }
+
+      ms.stop();
+
+      return {
+        jobId: jobInfo.id,
+        processedRecords: jobInfo.numberRecordsProcessed,
+        successfulRecords: jobInfo.numberRecordsProcessed - (jobInfo.numberRecordsFailed ?? 0),
+        failedRecords: jobInfo.numberRecordsFailed,
+      };
+    } catch (err) {
+      const jobInfo = await job.check();
+
+      // send last data update so job status/num. of records processed/failed represent the last update
+      ms.goto('Processing the job', jobInfo);
```

Comment:
if `job.poll` on L138 throws then the state rendered is from the previous poll update so we do one last update with fresh data from the org (`job.check()`) so the last state rendered is accurate (job state, record counter).